Client/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift

```diff
@@ -529,45 +529,87 @@ extension BrowserViewController: WKNavigationDelegate {
     return .allow
   }
 
-  // When this function is moved to Async-Await, do NOT forget to move the `TabManagerNavDelegate` to async-await!
-  public func webView(_ webView: WKWebView, didReceive challenge: URLAuthenticationChallenge, completionHandler: @escaping (URLSession.AuthChallengeDisposition, URLCredential?) -> Void) {
+  @MainActor
+  public func webView(_ webView: WKWebView, respondTo challenge: URLAuthenticationChallenge) async -> (URLSession.AuthChallengeDisposition, URLCredential?) {
```

Comment:
Do we need to execute this on MainActor?

Reply:
No but the old WebKit code with callbacks, calls it on `Main` like all the other WebKit delegates. So I made it the same as before and it matches all the other async-await delegates.

Reply:
Since we're changing a majority of the logic in here let's just dump the main actor isolation here and then dump the `Task.detached` part